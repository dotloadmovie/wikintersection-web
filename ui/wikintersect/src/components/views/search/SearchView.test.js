import React from 'react';
import {SearchViewContainer} from './SearchView.container';

import { shallow } from 'enzyme';

const props = {
    search0: {rowKey: null, value: null, search: null, results: null, serverRequestInFlight: false},
    search1: {rowKey: null, value: null, search: null, results: null, serverRequestInFlight: false},
    currView: 'search',
    index: 0,
    placeholder: "Search for a first Wikipedia entry"
}

it('displays correctly in default mode', () => {
    const component = shallow(<SearchViewContainer {...props}/>)

    expect(component).toMatchSnapshot();
})

props.search0.serverRequestInFlight = true;

it('displays correctly with a spinner', () => {
    const component = shallow(<SearchViewContainer {...props}/>)

    expect(component).toMatchSnapshot();
})


