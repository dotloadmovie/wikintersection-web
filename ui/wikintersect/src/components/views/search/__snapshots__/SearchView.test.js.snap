// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays correctly in default mode 1`] = `
<div
  className="search-container"
>
  <Search
    enterButton="Search"
    onChange={[Function]}
    onSearch={[Function]}
    placeholder="Search for a first Wikipedia entry"
    size="large"
    value={null}
  />
</div>
`;

exports[`displays correctly with a spinner 1`] = `
<div
  className="search-container"
>
  <Search
    enterButton="Search"
    onChange={[Function]}
    onSearch={[Function]}
    placeholder="Search for a first Wikipedia entry"
    size="large"
    value={null}
  />
  <div
    className="spin-position"
  >
    <Spin
      size="default"
      spinning={true}
      wrapperClassName=""
    />
  </div>
</div>
`;

exports[`displays properly with a result set 1`] = `
<div
  className="search-container"
>
  <Search
    enterButton="Search"
    onChange={[Function]}
    onSearch={[Function]}
    placeholder="Search for a first Wikipedia entry"
    size="large"
    value={null}
  />
  <Connect(SearchResultViewContainer)
    index={0}
    results={
      Array [
        "386BSD",
        "A/UX",
        "Android (operating system)",
        "Apache HTTP Server",
        "AudioOS",
        "BSD/OS",
        "BSD Daemon",
      ]
    }
  />
</div>
`;
